src/components/queue/QueueList/QueueList.tsx

```diff
@@ -1,89 +1,159 @@
-import React, {FC, useEffect, useState} from "react";
-import {
-    Box,
-    CircularProgress,
-    Grid,
-    Stack,
-    Typography,
-} from "@mui/material";
-import Button from "@components/shared/Button";
+
+import BouncingCubesAnimation from "@components/animations/BouncingCubesAnimation";
 import CreateTicketDialog from "@components/queue/CreateTicketDialog";
 import QueueListItem from "@components/queue/QueueListItem";
-import {useTickets} from "@util/queue/hooks";
-import {Queue, Ticket} from "@util/queue/api";
-import {useAuth} from "@util/auth/hooks";
-import BouncingCubesAnimation from "@components/animations/BouncingCubesAnimation";
+import Button from "@components/shared/Button";
+import {
+  Box,
+  Chip,
+  CircularProgress,
+  Divider,
+  Grid,
+  Stack,
+  Tooltip,
+  Typography
+} from "@mui/material";
+import { useAuth } from "@util/auth/hooks";
+import { Queue, Ticket, TicketStatus } from "@util/queue/api";
+import { useTickets } from "@util/queue/hooks";
+import { TICKET_COOLDOWN_MINUTES } from "@util/shared/constants";
+import React, { FC, useEffect, useState } from "react";
 
 export interface QueueListProps {
-    queue: Queue
-    showCompletedTickets: boolean;
+  queue: Queue;
+  showCompletedTickets: boolean;
 }
 
 /**
  * QueueList lists out the tickets in a queue.
  */
 const QueueList: FC<QueueListProps> = ({queue, showCompletedTickets}) => {
     const {currentUser, isTA} = useAuth();
-    const [tickets, ticketsLoading] = useTickets(queue.id, showCompletedTickets);
+    const [tickets, ticketsLoading] = useTickets(queue.id, false);
+    const shownTickets = tickets?.filter(t => !showCompletedTickets || t.status !== TicketStatus.StatusComplete);
     const [createTicketDialog, setCreateTicketDialog] = useState(false);
 
-    const inQueue = tickets && tickets.filter(ticket => ticket.user.Email == currentUser?.email).length > 0;
+    const [currentTime, setCurrentTime] = useState(new Date());
+
+    useEffect(() => {
+        const id = setInterval(() => {
+            setCurrentTime(new Date());
+        }, 5000);
+        return () => clearInterval(id);
+    });
+
+    const disabled = !tickets || tickets.some(t =>
+        (t.status === TicketStatus.StatusComplete) 
+        && (t.user.Email === currentUser!.email) 
+        && (currentTime.getTime() - t.completedAt!.toDate().getTime())/1000/60 <= (TICKET_COOLDOWN_MINUTES));
```

Comment:
is there no time.minutes() function? if not this is fine, maybe add parens

Reply:
Please use the `getMinutes` method for Date: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getMinutes